exec/java-exec/src/main/java/org/apache/drill/exec/physical/base/AbstractGroupScanWithMetadata.java

```diff
@@ -146,15 +148,23 @@ public boolean isMatchAllMetadata() {
    */
   @Override
   public long getColumnValueCount(SchemaPath column) {
-    long tableRowCount = (long) TableStatisticsKind.ROW_COUNT.getValue(getTableMetadata());
+    long tableRowCount, colNulls;
+    Long nulls;
     ColumnStatistics columnStats = getTableMetadata().getColumnStatistics(column);
-    long colNulls;
+    ColumnStatistics nonInterestingColStats = getNonInterestingColumnsMetadata().getColumnStatistics(column);
+
     if (columnStats != null) {
-      Long nulls = (Long) columnStats.getStatistic(ColumnStatisticsKind.NULLS_COUNT);
-      colNulls = nulls != null ? nulls : GroupScan.NO_COLUMN_STATS;
+      tableRowCount = (long) TableStatisticsKind.ROW_COUNT.getValue(getTableMetadata());
+    } else if (nonInterestingColStats != null) {
```

Comment:
Yes, tablemetadata holds only the interesting column stats. ColumnStats will be non-null only for the interesting columns. There are three scenarios here:
 
- Column is interesting. Then the columnStats is not null and it has the correct row count.
- Column is non-interesting. Then the columnStats will be null and nonInterestingColStats is not null and the row count is marked as unknown.
- Column doesn't exist in the table. So both columnStats and nonInterestingColStats will be null. Since the column doesn't exist, the row count is set to 0.

Here we know that a column will only belong to one of the above cases. Thats the reason there is an else-if instead of if.

Reply:
Since only one of `columnStats` (btw, let's call it `interestingColStats` to be consistent) or `nonInterestingColStats` will be non-null, we should ideally invoke `getNonInterestingColumnsMetadata()` only if the `columnStats` is null...otherwise the second call is redundant.  So, something like: 
  `ColumnStatistics interestingColStats = getTableMetadata().getColumnStatistics(column);`
   `ColumnStatistics nonInterestingColStats = null;`
    `if (interestingColStats == null) {`
        `nonInterestingColStats = getNonInterestingColumnsMetadata().getColumnStatistics(column);`
      `}`
Also, for the last `else` case, add a comment that the `return 0` is for non-existing columns.

Reply:
Changed and added a comment